### PR TITLE
LPS-133543 Adding new SAML test cases

### PR DESCRIPTION
**Pull request by ferenconodiD**
[LPS-133543](https://issues.liferay.com/browse/LPS-133543)

### CODE REVIEW

**Comment by ferenconodiD** [comment]
ci:test:sf

**Comment by ferenconodiD** [comment]
ci:test:security:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 1]
```diff
@@ -374,6 +374,63 @@ definition {
...
+	test AssertFirstNameCannotBeUsedForUserMatchingEvenWithLDAPEnabled {
+		property apacheds.enabled = "true";
+		property app.server.bundles.size = "0";
+		property test.name.skip.portal.instance = "SAML#AssertFirstNameCannotBeUsedForUserMatchingEvenWithLDAPEnabled";
+
+		PortalSettings.addLDAPServer(LDAPServerName = "ApacheDS Server");
+
+		PortalSettings.editLDAPServer(LDAPServerName = "ApacheDS Server");
+
+		PortalSettings.testLDAPConnection();
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:388]: SACRED: Could we combine these 3 LDAP macros into one?

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 2]
```diff
@@ -374,6 +374,63 @@ definition {
...
+
+		var enableSettingList = "enabled,ldapImportEnabled";
+
+		PortalSettings.configureLDAPSetting(enableSettingList = "${enableSettingList}");
+
+		PortalSettings.viewEnabledLDAPSetting(enableSettingList = "${enableSettingList}");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:394]: SACRED: Again, we could set a dedicated macro for applying this setup in these 3 steps

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 3]
```diff
@@ -374,6 +374,63 @@ definition {
...
+
+		Type(
+			locator1 = "TextInput#SAML_NAME_FIELD",
+			value1 = "samlidp");
+
+		Type(
+			locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_ENTITY_ID_FIELD",
+			value1 = "samlidp");
+
+		Click(locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_USER_RESOLUTION_ATTRIBUTE");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:419]: SACRED: We could combine types and clicks from test case level, and put them into a dedicated macro

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 4]
```diff
@@ -374,6 +374,63 @@ definition {
...
+
+		SAMLPortlet.addIdentityProviderConnectionAttributeMapping(idpAttributeMapping = "firstName");
+
+		Click(
+			key_index = "0",
+			locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_USE_TO_MATCH_USERS_RADIO");
+
+		Button.clickSave();
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:427]: SACRED: We could move this 3 steps into some dedicated macro

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 5]
```diff
@@ -537,6 +594,63 @@ definition {
 		}
 	}
 
+	@description = "LPS-123218 TC-3 Assertion for: last name cannot be used for user matching even if LDAP import is enabled."
+	@priority = "4"
+	test AssertLastNameCannotBeUsedForUserMatchingEvenWithLDAPEnabled {
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:599]: More or less, the same comments apply here as for the previous test.

**Review by gaborlovasdotcom — CHANGES_REQUESTED** [review]
I noted some change requests, most of them about combining elemental steps into one new macro. Even if we don't apply all SACRED changes, we should combine these steps.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase (hunk 1 above)
Comment (ferenconodiD) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:388]: Good point! Created LDAP.macro for it.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase (hunk 2 above)
Comment (ferenconodiD) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:394]: Good point! Created LDAP.macro for it.

**Comment by ferenconodiD** [comment]
ci:test:sf

**Comment by ferenconodiD** [comment]
ci:test:security:nocompile

**Comment by ferenconodiD** [comment]
ci:test:security:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 6]
```diff
@@ -374,6 +374,38 @@ definition {
 		AssertElementPresent(locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_ERROR_NO_MAPPING_SELECTED");
 	}
 
+	@description = "LPS-123218 TC-2 Assertion for: first name cannot be used for user matching even if LDAP import is enabled."
+	@priority = "4"
+	test AssertFirstNameCannotBeUsedForUserMatchingEvenWithLDAPEnabled {
+		property apacheds.enabled = "true";
+		property app.server.bundles.size = "0";
+		property portal.upstream = "feri";
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:382]: Please remove the custom property or revert it to true

**Review by gaborlovasdotcom — CHANGES_REQUESTED** [review]
Please remove the temp changes otherwise LGTM!

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase (hunk 6 above)
Comment (ferenconodiD) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase]: My bad! Removed already.

**Comment by weller0246** [comment]
LGTM

**Review by weller0246 — APPROVED** [review]

**Review by gaborlovasdotcom — APPROVED** [review]
LGTM

**Comment by gaborlovasdotcom** [comment]
ci:forward

**Pull request closed without merging**